src/core/api/project.go

```diff
@@ -712,3 +706,39 @@ func getProjectMemberSummary(projectID int64, summary *models.ProjectSummary) {
 
 	wg.Wait()
 }
+
+// Returns the highest role in the role list.
+// This func should be removed once we deprecate the "current_user_role_id" in project API
+// A user can have multiple roles and they may not have a strict ranking relationship
+func highestRole(roles []int) int {
+	if roles == nil {
+		return 0
+	}
+	rolePower := map[int]int{
+		common.RoleProjectAdmin: 50,
+		common.RoleMaster:       40,
+		common.RoleDeveloper:    30,
+		common.RoleGuest:        20,
+		common.RoleLimitedGuest: 10,
+	}
+	var ind int
+	for _, r := range roles {
+		if _, ok := rolePower[r]; ok {
+			roles[ind] = r
+			ind++
+		}
+	}
```

Comment:
This will change the input value of `roles`, this is not a good practice.

If only want to ensure that the value in roles is valid, it can process in line 737.

```go
p, ok := rolePower[role]
if ok && p > highestPower {
        highest = role
        highestPower = p
}
```

Reply:
I was thinking it's the right thing to do because it filters out the illegal values if any, but yeah ...